Fix IdentifyMissingTests tool

Androbin opened the pull request:
Initial:
* `FindTestsInFile` parses `it` but not `describe`
* * common names such as `should work` get mingled
* `IdentifyMissingTests` enumerates existing `PlaywrightTest` attributes, then compares them with upstream tests
* * finds `PlaywrightTest` attributes without corresponding upstream test
* * does not find upstream tests without corresponding `PlaywrightTest` attribute
* `Found/Mismatched/Missing: 703/565/56 out of 1324`

Fixed `missing-tests` tool:
* `FindTestsInFile` parses both `it` and `describe` (and associates `it` with the innermost surrounding `describe`)
* `IdentifyMissingTests` enumerates upstream tests, then compares them with existing `PlaywrightTest` attributes
* * finds upstream tests without corresponding `PlaywrightTest` attribute
* `Found/Mismatched/Missing: 578/572/645 out of 1795`
* Commits originally made in https://github.com/hardkoded/puppeteer-sharp/pull/1663

Fixed `PlaywrightTest` attributes:
* `Found/Mismatched/Missing: 703/583/509 out of 1795`

Androbin commented:
The 2 failing `PageNetworkIdleTests` are unrelated to the changeset

avodovnik reviewed (commented):
Thanks for the PR @Androbin. I'm curious about the differences in the number of found tests. How come that number changed between the PRs - in theory, it shouldn't have.

Androbin commented on src/tools/PlaywrightSharp.Tooling/IdentifyMissingTests.cs:
```diff
@@ -65,27 +65,28 @@ public static void Run(IdentifyMissingTestsOptions options)
                 // a test can either be a full match, a partial (i.e. just the test name) or no match
-                var potentialMatch = _testPairs.Where(x => string.Equals(x.TestName, atx.TestName, StringComparison.InvariantCultureIgnoreCase));
+                var potentialMatch = attributes.Where(atx => string.Equals(x.TestName, atx.TestName, StringComparison.InvariantCultureIgnoreCase))
+                                               .Where(atx => string.Equals(x.Describe, atx.Describe, StringComparison.InvariantCultureIgnoreCase));
```
@avodovnik Without this additional `.Where` term:
`Found/Mismatched/Missing: 695/585/515 out of 1795`

Androbin replied:
The other 8 are due to one upstream test being attributed to multiple `PlaywrightTest` attributes.

kblok replied:
We don't need two wheres here. It could be only one with an `&&`

kblok replied:
I think it would be great if you can find one example so we can check what's right or wrong.

Androbin commented:
@avodovnik As you can see in the last commit, the PR uncovered a number of bad attributes that have now been fixed:
`Found/Mismatched/Missing: 703/583/509 out of 1795`

Androbin commented:
Initial:
`Found/Mismatched/Missing: 703/565/56 out of 1324`

Fixed `missing-tests` tool:
`Found/Mismatched/Missing: 578/572/645 out of 1795`

Fixed `PlaywrightTest` attributes:
`Found/Mismatched/Missing: 703/583/509 out of 1795`

mxschmitt commented:
Closing since it seems not needed anymore and the whole test infrastructure has changed.
